navis/nbl/smat.py

```diff
@@ -247,37 +268,55 @@ def _query_many(self, idx_pairs, threads=None) -> Iterator[List[np.ndarray]]:
         chunks = chunksize(len(idx_pairs), threads)
 
         with ProcessPoolExecutor(threads) as exe:
-            yield from exe.map(
-                self._query, idx_pairs[:, 0], idx_pairs[:, 1], chunksize=chunks
-            )
+            yield from exe.map(self.match_fn,
```

Comment:
Mapping `self._query` turned out to be a bottleneck (both for speed and memory footprint). I _think_ that's because the entire builder including several thousand neurons had to be serialised. Instead, we're now operating directly on the matching function and sending only the pairs we're actually interested in. This seems to improve things but I still get less performance than expected.

Reply:
I could try pathos which uses dill and see if that improves things?